internal/subscription/validation.go

```diff
@@ -23,8 +25,68 @@ const (
 
 const (
 	// List of patchable attributes(proto)
-	pushConfigPath     = "push_config"
-	ackDeadlineSecPath = "ack_deadline_seconds"
+	pushConfigPath       = "push_config"
+	ackDeadlineSecPath   = "ack_deadline_seconds"
+	retryConfigPath      = "retry_policy"
+	deadLetterConfigPath = "dead_letter_policy"
+)
+
+// Config validations related values
+const (
```

Comment:
the consts and the error vars defined here seem to be relavent only inside the subscription package. Do you think it would be better if we make these constants private instead of public? i.e names starting with small case.

Reply:
if we are writing functional tests outside of package it might be usefule to use the same constants in tests while testing. 
Today our errors are not 100% compatible with google pubsub, we will need to relook at all our error codes. can take this up then.